File: android/cardemory/carddata/src/main/java/com/cardemory/carddata/entity/Card.kt
package com.cardemory.carddata.entity

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Card(
    val id: Long = UNDEFINED_ID,
    val cardSetId: Long,
    val title: String,
    val description: String,
    val memoryRank: Double,
    val lastTrainMillis: Long
) : Parcelable {

    constructor(
        title: String,
        description: String
    ) : this(UNDEFINED_ID, UNDEFINED_ID, title, description, 0.0, 0)

    constructor() : this(UNDEFINED_ID, UNDEFINED_ID, "", "", 0.0, 0)

    companion object {
        const val UNDEFINED_ID = -1L
    }
}

File: flutter/cardemory/lib/domain/card/entity/card.dart
import 'package:equatable/equatable.dart';

class Card extends Equatable {
  static const unknownId = -1;

  final int id;
  final int cardSetId;
  final String title;
  final String description;
  final double memoryRank;
  final int lastTrainingMillis;

  const Card(
    this.cardSetId,
    this.title,
    this.description,
    this.memoryRank,
    this.lastTrainingMillis, {
    this.id = unknownId,
  });

  Card copyWith({
    final int? id,
    final int? cardSetId,
    final String? title,
    final String? description,
    final double? memoryRank,
    final int? lastTrainingMillis,
  }) =>
      Card(
        cardSetId ?? this.cardSetId,
        title ?? this.title,
        description ?? this.description,
        memoryRank ?? this.memoryRank,
        lastTrainingMillis ?? this.lastTrainingMillis,
        id: id ?? this.id,
      );

  @override
  List<Object> get props => [id, title, description];
}
